test/E2ETest/TyeBuildTests.Dockerfile.cs

```diff
@@ -86,5 +87,44 @@ public async Task TyeBuild_SinglePhase_ExistingDockerfile()
                 await DockerAssert.DeleteDockerImagesAsync(output, imageName);
             }
         }
+
+        [ConditionalFact]
+        [SkipIfDockerNotRunning]
+        public async Task TyeBuild_SinglePhase_ExistingDockerfileWithBuildArgs()
+        {
+            var projectName = "single-phase-dockerfile-args";
+            var environment = "production";
+            var imageName = "test/web";
+
+            await DockerAssert.DeleteDockerImagesAsync(output, imageName);
+
+            using var projectDirectory = CopyTestProjectDirectory(projectName);
+            Assert.True(File.Exists(Path.Combine(projectDirectory.DirectoryPath, "Dockerfile")), "Dockerfile should exist.");
+
+
+            var projectFile = new FileInfo(Path.Combine(projectDirectory.DirectoryPath, "tye.yaml"));
+
+            var outputContext = new OutputContext(sink, Verbosity.Debug);
+            var application = await ApplicationFactory.CreateAsync(outputContext, projectFile);
+
+            application.Registry = new ContainerRegistry("test");
+
+            try
+            {
+                await BuildHost.ExecuteBuildAsync(outputContext, application, environment, interactive: false);
+
+                Assert.Single(application.Services.Single().Outputs.OfType<DockerImageOutput>());
+                var builder = (DockerFileServiceBuilder)application.Services.First();
+                var valuePair = builder.BuildArgs.First();
+                Assert.Equal("pat", valuePair.Key);
```

Comment:
Some other things you could test here:
- What happens when duplicated build args are passed in?
- Passing in multiple args

Reply:
what's the behaviour that you think is valid for duplicate args?  take the first of the duplicates, or fail gracefully?

Reply:
Added multiple parameter functionality and test